### PR TITLE
add user and date to provenance

### LINKED ISSUE #14: Include date and username in provenance metadata?
Generation date and username can be very useful for knowing which versions of a file are more up to date and  knowing who to ask about it. Could these be added to `get_provenance_metadata`?

(I know date is somewhat redundant since it's sort-of implicit in the git hash, but it's nicer to have something more direct.)

### PR DESCRIPTION
**Pull request by anton-seaice**
Added user & date to the `get_provenance_metadata` function (closes #14)

There's a little duplication with each script now, but I didn't remove where user / time is added twice.

I ran 

`python3 mesh_generation/generate_mesh.py --grid-type mom --grid-filename /g/data/ik11/inputs/access-om2/input_20230515_025deg_topog/mom_025deg/ocean_hgrid.nc --mesh-filename test.nc`

and it looks good:
`
:history = "Created by as2285, on 2024-07-19, using https://github.com/COSIMA/om3-scripts/blob/bec68e031beaa62dabe2f87e8c0401b9adac3306/mesh_generation/generate_mesh.py: python3 generate_mesh.py --grid-type=mom --grid-filename=/g/data/ik11/inputs/access-om2/input_20230515_025deg_topog/mom_025deg/ocean_hgrid.nc --mesh-filename=/g/data/tm70/as2285/om3-scripts/test.nc" ;`

### CODE REVIEW

File: scripts_common.py [hunk 1]
```diff
@@ -84,6 +85,8 @@ def get_provenance_metadata(file, runcmd):
         runcmd: the command used to run the file (with any arguments)
     """
 
+    prepend = f"Created by {os.environ['USER']}, on {datetime.now().strftime('%Y-%m-%d')}, using "
```
Comment (aekiss) [inline scripts_common.py:88]:
```suggestion
    prepend = f"Created by {os.environ['USER']} on {datetime.now().strftime('%Y-%m-%d')}, using " 
```

Comment (aekiss) [inline scripts_common.py:88]: > os.environ['USER']

Is there a way to put the actual name in parentheses after this? Maybe using `git config --global user.name` if it exists?

File: scripts_common.py [hunk 2]
```diff
@@ -96,13 +99,13 @@ def get_provenance_metadata(file, runcmd):
...
-        prepend = f"Created using {git_url}: "
+        prepend += f"{git_url}: "
     else:
         warn(
-            f"{file} not under git version control! Add you file to a repository before generating any production outpout."
+            f"{file} not under git version control! Add your file to a repository before generating any production output."
         )
-        prepend = f"Created using {file}: "
-
+        prepend += f"Created using {file}: "
```
Comment (aekiss) [inline scripts_common.py:107]:
```suggestion
        prepend += f"{file}: "
```

**Review by aekiss — COMMENTED** [review]
Thanks Anton, looks good, just a few suggestions.

**Comment by aekiss** [comment]
refer to issue https://github.com/COSIMA/om3-scripts/issues/14

**Comment by anton-seaice** [comment]
Updated example output:

`:history = "Created by anton-seaice on 2024-07-26, using https://github.com/COSIMA/om3-scripts/blob/b3a740db8909686da97d7a38d4809004ea92a3a3/mesh_generation/generate_mesh.py: python3 generate_mesh.py --grid-type=mom --grid-filename=/g/data/ik11/inputs/access-om2/input_20230515_025deg_topog/mom_025deg/ocean_hgrid.nc --mesh-filename=/g/data/tm70/as2285/om3-scripts/test.nc"`

File: scripts_common.py (hunk 1 above)
Comment (anton-seaice) [reply scripts_common.py]: Ok done :)

**Comment by anton-seaice** [comment]
Back to you @aekiss

File: scripts_common.py [hunk 3]
```diff
@@ -76,6 +76,24 @@ def git_status(file):
...
+    dirname = os.path.dirname(file)
+
+    try:
+        name = (
+            subprocess.check_output(["git", "-C", dirname, "config", "user.name"])
+            .decode("ascii")
+            .strip()
+        )
+    except subprocess.CalledProcessError:
+        name = os.environ["USER"]
```
Comment (aekiss) [inline scripts_common.py:92]: How about this so we always get NCI username? NB: untested
```suggestion
    name = os.environ["USER"]
    try:
        name += 
            "(" +
            (
                subprocess.check_output(["git", "-C", dirname, "config", "user.name"])
                .decode("ascii")
                .strip()
            )
            + ")"
    except subprocess.CalledProcessError:
        pass
```

**Comment by anton-seaice** [comment]
Third time lucky example output:

`:history = "Created by as2285 (anton-seaice) on 2024-07-29, using https://github.com/COSIMA/om3-scripts/blob/e12d19334baf4dfa73d44291b3aeae7cdf2db74c/mesh_generation/generate_mesh.py: python3 generate_mesh.py --grid-type=mom --grid-filename=/g/data/ik11/inputs/access-om2/input_20230515_025deg_topog/mom_025deg/ocean_hgrid.nc --mesh-filename=/g/data/tm70/as2285/om3-scripts/test.nc" ;`

File: scripts_common.py (hunk 3 above)
Comment (anton-seaice) [reply scripts_common.py]: Ok i've done this now. Back to you @aekiss

**Comment by anton-seaice** [comment]
@aekiss Can you have another look please ?

**Review by aekiss — APPROVED** [review]
LGTM, thanks @anton-seaice

**Pull request merged**